pandas/tests/plotting/test_xlim_ylim_dtype.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import pandas as pd
+from datetime import datetime as dt
+import numpy as np
+
+"""targeting #GH40781"""
+
+# arbitrary df to test
+df = pd.DataFrame(
+    {
+        "A" : [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        "B": [dt.now() for i in range(0, 10)]
+    }
+)
+
+
+def test_axis_lim_invalid():
+    global df
```

Comment:
I suggest that you do not use ``global df``.
Instead, you may want to make a fixture returning your dataframe, and passing this fixture into your test functions.
Alternatively, you may create a class with your test methods and define an attribute with the dataframe (``self.df``), which you can reuse in the test methods.

Reply:
I used parametrization in pytest to create custom dataframe and ended up not having to use globals at all.

Reply:
Please see: https://github.com/pandas-dev/pandas/pull/40889/commits/e3b69dfd2e129558e790ae3398e6a8caaef1a4a3

Reply:
My bad. I was looking at an outdated version.
Then it is not clear for me why the tests are failing.